File: 111. Minimum Depth of Binary Tree/111. Minimum Depth of Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int MinDepth(TreeNode root) {
        if (root == null) return 0;
        if (root.left == null && root.right == null) return 1;
        
        int leftDepth = root.left != null ? MinDepth(root.left) : int.MaxValue;
        int rightDepth = root.right != null ? MinDepth(root.right) : int.MaxValue;
        
        return 1 + Math.Min(leftDepth, rightDepth);
    }
}

File: 111. Minimum Depth of Binary Tree/111. Minimum Depth of Binary Tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */
function minDepth(root: TreeNode | null): number {
    if (!root) return 0;
    if (!root.left && !root.right) return 1;
    
    const leftDepth = root.left ? minDepth(root.left) : Number.MAX_VALUE;
    const rightDepth = root.right ? minDepth(root.right) : Number.MAX_VALUE;
    
    return 1 + Math.min(leftDepth, rightDepth);
}